packages/shared/friends/sagas.ts

```diff
@@ -198,14 +200,24 @@ function* configureMatrixClient(action: SetMatrixClient) {
   }
 
   // initialize conversations
-  client.onStatusChange((socialId, status) => {
+  client.onStatusChange(async (socialId, status) => {
     const userId = parseUserId(socialId)
     if (userId) {
+      // When it's a friend and is not added to catalog
+      // unity needs to know this information to show that the user has connected
+      if (isFriend(store.getState(), userId) && !isAddedToCatalog(store.getState(), userId)) {
+        await ensureFriendProfile(userId)
+        getUnityInstance().AddFriends({
+          friends: [userId],
+          totalFriends: getTotalFriends(store.getState())
+        })
+      }
+
       sendUpdateUserStatus(userId, status)
     }
   })
 
-  client.onMessage((conversation, message) => {
+  client.onMessage(async (conversation, message) => {
```

Comment:
who catches the errors of this async function?

Reply:
Refactored the function so that it will never throw an error